src/java/org/sogive/data/loader/ImportEWCCData.java

```diff
@@ -0,0 +1,154 @@
+package org.sogive.data.loader;
+
+import java.io.BufferedReader;
+import java.io.File;
+
+import org.sogive.data.charity.NGO;
+import org.sogive.data.charity.SoGiveConfig;
+import org.sogive.data.charity.Thing;
+
+import com.winterwell.data.KStatus;
+import com.winterwell.depot.Depot;
+import com.winterwell.depot.Desc;
+import com.winterwell.es.ESPath;
+import com.winterwell.gson.JsonArray;
+import com.winterwell.gson.JsonElement;
+import com.winterwell.gson.JsonObject;
+import com.winterwell.gson.JsonParser;
+import com.winterwell.maths.stats.distributions.discrete.ObjectDistribution;
+import com.winterwell.utils.Dep;
+import com.winterwell.utils.Utils;
+import com.winterwell.utils.containers.ArrayMap;
+import com.winterwell.utils.io.FileUtils;
+import com.winterwell.utils.log.KErrorPolicy;
+import com.winterwell.web.ajax.JThing;
+import com.winterwell.web.app.AppUtils;
+
+/**
+ * Charity Commission for England & Wales. This can upload the _basic_ data from
+ * their register into SoGive. TODO upload more of what they hold
+ * 
+ * @author sarah
+ *
+ */
+public class ImportEWCCData {
+
+	public static final String EWCC_REG = "englandWalesCharityRegNum";
+	
+	private volatile boolean running;
+
+	public boolean isRunning() {
+		return running;
+	}
+	
+	private static void doCreateCharity(NGO ngo) {
+
+		ngo.put("uk_giftaid", true);
+
+		SoGiveConfig config;
+		if (Dep.has(SoGiveConfig.class)) {
+			config = Dep.get(SoGiveConfig.class);
+		} else {
+			config = AppUtils.getConfig("sogive", new SoGiveConfig(), null);
+		}
+
+		String ourId = ngo.getId();
+		if (Utils.isBlank(ourId)) {
+			ourId = NGO.idFromName(ngo.getName());
+			ngo.put(ngo.ID, ourId);
+		}
+		ESPath draftPath = config.getPath(null, NGO.class, ourId, KStatus.DRAFT);
+		ESPath publishPath = config.getPath(null, NGO.class, ourId, KStatus.PUBLISHED);
+
+		JThing item = new JThing().setJava(ngo);
+		//AppUtils.doSaveEdit(draftPath, item, null, null);
+		AppUtils.doPublish(item, draftPath, publishPath);
+		
+	}
+	
+	private static String jsonToString(JsonObject jObject, String label) {
+	
+		JsonElement jElement = jObject.get(label);
+		return (jElement.isJsonNull()) ? null : jElement.getAsString();
+		
+	}
+
+	private static Desc<File> getDesc() {
+		Desc desc = new Desc("json.ewcc.charities.zip", File.class);
```

Comment:
@sarahboka Could you add a javadoc comment to document how to get this file? Thanks

Reply:
Actually, as I'm looking into it, I think this should be changed. I've been running it twice. The first time, it hits the catch block in Depot class get2() function causing it to open a file selector in the GUI where I can upload the file to the Depot. The second time, it runs. 
But I'm going to upload the file without the file selector gui. Will start working on that & add javadocs too.